src/v/cluster/tm_stm.cc

```diff
@@ -28,6 +28,19 @@
 #include <ranges>
 namespace cluster {
 
+namespace {
+template<typename T>
+ss::future<checked<T, tm_stm::op_status>>
+shutdown_exception_handler(const std::exception_ptr& e) {
+    if (ssx::is_shutdown_exception(e)) {
+        // map shutdown exception to timeout error as we can not be
+        // certain about the operation outcome.
+        return ssx::now<checked<T, tm_stm::op_status>>(
+          tm_stm::op_status::timeout);
+    }
+    return ss::make_exception_future<checked<T, tm_stm::op_status>>(e);
+}
```

Comment:
nit: if this function accepted a future as an argument it could be used without specifying return type.
Something like this:
```
template<typename F>
handle_shutdown_exceptions(F&& fut) {
    return std::move(fut).handle_exception([](const std::exception_ptr& e) {
       using fut_value_t = F::value_type;
       ...
       return ssx::now<fut_value_t>(tm_stm::op_status::timeout);
       ...
    };
}
```

and use like:
```
return handle_shutdown_exceptions(ss::with_gate(...));
```